### PR TITLE
Fix #63204: Make proficiency account for SKILL_TRAINING_SPEED

### LINKED ISSUE #63204: skill training speed should also affect proficiency xp
### Is your feature request related to a problem? Please describe.

skill training speed affects xp from both training and books but not proficiency.  with  skill training speed either 0 or 100 multiplier proficiency xp gain is same

### Solution you would like.

skill training speed should also affect proficiency xp 

### Describe alternatives you have considered.

maybe add a seperate profeciency xp gain multiplier on debug settings

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kromgart**
#### Summary
Bugfixes "Make proficiency account for SKILL_TRAINING_SPEED"

#### Purpose of change

Fix #63204 

#### Describe the solution

Proficiency has a field _time_to_learn. I checked the code and seems like this field is accessed only when creating proficiency from json, and then read only through corresponding getter time_to_learn(). I modified the getter, so that the value is adjusted accordingly to the SKILL_TRAINING_SPEED game option.

#### Testing

New test char. Made 5 Z corpses, dissected them with scalpel, one time with default SKILL_TRAINING_SPEED (1.0), and then with modified (0.2). Numbers seem to respect the option now, see screenshots.

SKILL_TRAINING_SPEED 0.2:

prof1a

prof1b

SKILL_TRAINING_SPEED 1.0:

prof2a

prof2b

#### Additional context

UPD: fixed screenshots order

### CODE REVIEW

**Comment by alp7292** [comment]
that was quick thanks for your efforts

**Comment by Kromgart** [comment]
> That was quick thanks for your efforts does this also works with bigger numbers like 2.0 or 5.0. Skill training speed can be be between 0.0 to 100.0

Yes, prof's _time_to_learn will become smaller (less hours needed) if you use bigger number.

File: src/proficiency.cpp [hunk 1]
```diff
@@ -179,7 +179,12 @@ float proficiency::default_weakpoint_penalty() const
 
 time_duration proficiency::time_to_learn() const
 {
-    return _time_to_learn;
+    const double scaling = get_option<float>( "SKILL_TRAINING_SPEED" );
+    if( scaling != 1.0 && scaling != 0.0 ) {
```
Comment (Fris0uman) [inline src/proficiency.cpp:183]: Is the 0.0 option also ignored in the other uses of the setting? Shouldn't 0 mean no learning?

**Comment by intergalacticdata** [comment]
Makes sense to me.

**Comment by alp7292** [comment]
if we remove the "&& scaling != 0.0" would that work

**Comment by alp7292** [comment]
if we remove the "&& scaling != 0.0" would that work

File: src/proficiency.cpp (hunk 1 above)
Comment (alp7292) [reply src/proficiency.cpp:183]: if we remove the "&& scaling != 0.0" would that work

Comment (alp7292) [reply src/proficiency.cpp:183]: yes 0 should mean no learning it shouldnt ignored

Comment (Kromgart) [reply src/proficiency.cpp:188]: Fixed with the next commit:  when the option is 0.0 there is no progress on proficiency anymore and the UI is more consistent.

File: src/proficiency.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "color.h"
 #include "flat_set.h"
 #include "optional.h"
+#include "options.h"
```
Comment (irwiss) [inline src/proficiency.h:15]: If the stuff from options.h is only used in the cpp file it's best to move it there, so it doesn't get included everywhere proficiency.h is included

**Comment by alp7292** [comment]
now it fails great

**Pull request merged**